python/ray/serve/_private/common.py

```diff
@@ -118,17 +118,56 @@ class DeploymentStatusTrigger(str, Enum):
     DELETING = "DELETING"
 
 
+# Internal Enum used to manage deployment state machine
+class DeploymentStatusTransition(str, Enum):
+    HEALTHY = "HEALTHY"
+    CONFIG_UPDATE = "CONFIG_UPDATE"
+    AUTOSCALE_UP = "AUTOSCALE_UP"
+    AUTOSCALE_DOWN = "AUTOSCALE_DOWN"
+    MANUALLY_INCREASE_NUM_REPLICAS = "MANUALLY_INCREASE_NUM_REPLICAS"
+    MANUALLY_DECREASE_NUM_REPLICAS = "MANUALLY_DECREASE_NUM_REPLICAS"
+    REPLICA_STARTUP_FAILED = "REPLICA_STARTUP_FAILED"
+    HEALTH_CHECK_FAILED = "HEALTH_CHECK_FAILED"
+    INTERNAL_ERROR = "INTERNAL_ERROR"
+    DELETE = "DELETE"
```

Comment:
So transition is a tuple of `(From State, To State, Trigger)`. 

Occasionally it's trivialized to just refer to a trigger (in case it identifies it uniquely), but i'd still suggest we rather call it trigger (to disambiguate w/ existing one, we can call it `_DeploymentStatusInternalTrigger`)